README.md

```diff
@@ -11,13 +11,58 @@ SnapShell is a powerful command-line utility designed to enhance your Linux term
 
 ## Installation
 
-Install SnapShell via pip with the following command:
+### Prerequisites
+- Python 3.7 or higher
+- `pip` package manager
+
+### Installing via pip 
+Install SnapShell with the following command:
 
 ```sh
 pip install snapshell
 ```
+### Custom Installation
+SnapShell requires your `GROQ_API_KEY` during installation. Follow the on-screen instructions after installation
+
+The installation will automatically detect your shell (bash, zsh, fish) and set the key in the respective configuration file.
+
+OR
+
+```sh
+git clone https://github.com/KTS-o7/snapshell.git
+cd snapshell
+python -m venv env
+source env/bin/activate
+pip install -r requirements.txt
+python setup.py install
+```
+## Dependencies
 
-## Usage
+SnapShell requires the following libraries:
+
+- `argparse`: For parsing command-line arguments.
+- `colorama`: For colorized terminal output.
+- `sqlite3`: For local database operations.
+- `os`: To interact with your operating system.
+
+## Configuration Instructions
+To use SnapShell, you need to set up the `GROQ_API_KEY` environment variable. This is required for connecting to the underlying Language Model API.
+### Setting the `GROQ_API_KEY`
+1. Obtain your API key from the GROQ service.
+1. Set it as an environment variable:
+```sh
```

Comment:
we need to export as HELPER_GROQ_API_KEY. Read through the code base for details.